lib/client.js

```diff
@@ -962,7 +962,10 @@ MqttClient.prototype._handlePublish = function (packet, done) {
 
   switch (qos) {
     case 2:
-      this.incomingStore.put(packet, function () {
+      this.incomingStore.put(packet, function (err) {
+        if (err) {
+          return done && done(err)
```

Comment:
can done be null? If that is true, than can you please assign it to a noop function?

Reply:
I'm not sure but the existing code https://github.com/mqttjs/MQTT.js/blob/master/lib/client.js#L974 cheking it. So I applied the same manner.

I've checked call sequence.
`handlePublish` is called only from https://github.com/mqttjs/MQTT.js/blob/master/lib/client.js#L336 in `handlePacket`. And `handlePacket` is called only from https://github.com/mqttjs/MQTT.js/blob/master/lib/client.js#L292.
At that point, `nextTickWork` is passed as the 2nd parameter, callback.

`nextTickWork` is defined  https://github.com/mqttjs/MQTT.js/blob/master/lib/client.js#L283. It cannot be null.

So null checking is not required. I will remove it from my update and existing code.

Reply:
I removed `done &&` on my  environment.
Then I got errors on the following test:
https://github.com/mqttjs/MQTT.js/blob/master/test/abstract_client.js#L953

Can I remove the test?
Or revert as follows? 
```js
return done && done(err)
```

Reply:
Or

```js
MqttClient.prototype._handlePublish = function (packet, done = function (){}) {
```

?

Reply:
I choose

```js
MqttClient.prototype._handlePublish = function (packet, done = function (){}) {
```

Because

1. Someone might depend on existing test. So I shouldn't remove it.
2. You mentioned using empty function is better for empty callback at https://github.com/mqttjs/MQTT.js/pull/864#discussion_r213572395

Reply:
To support ES5 and older, I fixed as follows:

```js
MqttClient.prototype._handlePublish = function (packet, done) {
  done = typeof done !== 'undefined' ? done : function () {}
```

Reply:
There is a handy `nop()` function https://github.com/mqttjs/MQTT.js/blob/master/lib/client.js#L66.

Reply:
I didn't notice it. I replaced it.